department-of-veterans-affairs/caseflow
# Paula/Caseflow-200-add-edit-reasons

Resolves [Caseflow-200](https://vajira.max.gov/browse/CASEFLOW-200)

### Description
Caseflow-200 "Add Edit Reasons" adds a dropdown option in the "Edit NOD Date" modal, so that the user can select a reason for editing the date.  There are 2 options:  New Form/Information Received and Data Entry Error. The user must select an option in order to modify the NOD date.  The submit button on the modal is inactive until a valid date and a reason is selected.  In addition, a previous user story created a new table (nod_date_updates) and this user story populates that new table (here are the fields: id, appeal_id, old_date, new_date, user_id, change_reason, created_at, updated_at) and also updates the appeals table's receipt_date field.

### Acceptance Criteria
- [x] Reason dropdown displays a default placeholder and 2 values (New Form/Information Received, Data Entry Error).
- [x] Submit button is disabled until the user has a valid date and a reason selected.

### Testing Plan
1. Login as an approved user type (like BVAAABSHIRE)
2. Go to Queue and Search cases.
3. Find a case that has an existing NOD (like 123456789)
4. Goto the Case details page, scroll to the "Edit NOD Date" link
5. Click the link to launch the EditNodDate modal.
6. The "Submit" button is disabled upon initial opening.
7. There is now a dropdown for "Reason for edit". There should be 2 reasons: "New Form/Information Received" and "Data Entry Error."  Choose one.
8. If the NOD date selected in valid and a reason is selected, the "Submit" button will be enabled.
9. Clicking "Submit" will results in a success alert at the top of the page (no changes made there in this user story).
10. Clicking "Cancel" instead of "Submit" will reset the items in the modal to the default.
11. If an invalid NOD date is selected, the user is prompted as to why (not new functionality) and the Submit button remains disabled until the date is fixed.


### User Facing Changes
 - [ ] Screenshot of Edit NOD Date modal before the new dropdown
 - [ ] Screenshot of Edit NOD modal with the new "Reason" dropdown

 BEFORE|AFTER
 ---|---
![BeforeScreenshot](https://user-images.githubusercontent.com/23639439/105373877-201f0e80-5bd5-11eb-9dc6-e1f51b42acc1.jpg)
![afterScreenshot](https://user-images.githubusercontent.com/23639439/105373935-2d3bfd80-5bd5-11eb-9f41-ae3defc0140e.jpg)
